src/mango/test/16-index-selectors-test.py

```diff
@@ -189,6 +189,7 @@ def test_uses_partial_index_with_non_indexable_selector(self):
 @unittest.skipUnless(mango.has_text_service(), "requires text service")
 class IndexSelectorText(mango.DbPerClass):
     def setUp(self):
+        self.db = mango.Database(f"{mango.random_db_name()}_{str(self).split()[0]}")
```

Comment:
There are many tests files which use `self.db.recreate()` did you consider modifying that function?

Reply:
If using the same db name, `0.13` is the shortest time I need to avoid `500` locally.
This approach also increased the overall Mango test time to `60.730s`, compared to `49.936s` using custom database name.

```diff
def recreate(self):
-    NUM_TRIES = 10
+    NUM_TRIES = 3

-    for k in range(NUM_TRIES):
+    for k in range(1, NUM_TRIES):
         r = self.sess.get(self.url)
         if r.status_code == 200:
             db_info = r.json()
             docs = db_info["doc_count"] + db_info["doc_del_count"]
                 if docs == 0:
                     # db exists and it is empty -- exit condition is met
                     return
                 self.delete()
+        time.sleep(k * 0.13)
         self.create()
-        time.sleep(k * 0.1)
     raise Exception(
         "Failed to recreate the database after {} tries".format(NUM_TRIES)
     )
```

Reply:
All time based approaches are flowed. They don't solve the problem. They introduce flakiness. The options I see:

1. restructure all tests so they are not using recreate. They would always start from scratch and create fresh databaase with uniq name. 
2. update dreyfus_index_manager.erl to handle `cleanup` differently
3.  modify clouseau so it handle rapid re-creation better
4. change recreate function so it uses new name on each retry

Reply:
I would vote for option 4.

Reply:
Virtually, that is still about recreating the database, in my opinion.  But I agree that `create_random` would capture better the adjusted semantics for option 4.